src/main/java/com/walmartlabs/x12/standard/txset/asn856/DefaultAsn856TransactionSetParser.java

```diff
@@ -595,26 +595,30 @@ private void doBatchSegments(X12Segment segment, SegmentIterator segmentIterator
      * 
      * @param segments
      * @param txSet
-     * @throws X12ParserException if no HL loop is found
      */
     protected void parseSegmentsBeforeFirstLoop(SegmentIterator segments, AsnTransactionSet txSet) {
         
         while (segments.hasNext()) {
             X12Segment currentSegment = segments.next();
+            String segmentId = currentSegment.getIdentifier();
+            LOGGER.debug(segmentId);
             if (X12ParsingUtil.isHierarchalLoopStart(currentSegment)) {
+                // start of loops
                 // we should back up so 
                 // the parser starts w/ this segment
                 segments.previous();
                 break;
+            } else if (X12TransactionSet.TRANSACTION_SET_TRAILER.equals(segmentId)) {
+                // reached end of the transaction
+                // unexpected but let's back up
+                // and return to template
+                segments.previous();
+                break;
             } else {
                 // add DTM segments to the transaction set
                 // ignore other segments until we find the first loop
-                String segmentId = currentSegment.getIdentifier();
-                LOGGER.debug(segmentId);
                 if (DTMDateTimeReference.IDENTIFIER.equals(segmentId)) {
                     txSet.addDTMDateTimeReference(DTMDateTimeReferenceParser.parse(currentSegment));
-                } else {
-                    throw X12ParsingUtil.handleUnexpectedSegment(X12Loop.HIERARCHY_LOOP_ID, segmentId);
```

Comment:
was debating adding the unexpected segment to a list
`List<X12Segment> unexpectedSegmentsBeforeLoop;`

Reply:
more I thought about it, the more it makes sense to make these segments available... update coming soon